eidangservices/utils/strict.py

```diff
@@ -0,0 +1,329 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# EDIA NG webservices is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# ----
+#
+# Copyright (c) Sven Marti (ETH), Daniel Armbruster (ETH), Fabian Euchner (ETH)
+#
+# REVISION AND CHANGES
+# 2018/06/05        V0.1    Sven Marti
+# =============================================================================
+"""
+Keywordparser facilities for EIDA NG webservices.
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from builtins import * # noqa
+
+import functools
+import inspect
+
+from webargs.flaskparser import parser as flaskparser
+from marshmallow import Schema, exceptions
+
+from eidangservices import settings
+from eidangservices.utils.error import Error
+
+
+class KeywordParserError(Error):
+    """Base KeywordParser error ({})."""
+
+
+class ValidationError(KeywordParserError, exceptions.ValidationError):
+    """ValidationError: {}."""
+    pass
+
+
+def _callable_or_raise(obj):
+    """
+    Makes sure an object is callable if it is not ``None``. If not
+    callable, a ValueError is raised.
+    """
+    if obj and not callable(obj):
+        raise ValueError("{0!r} is not callable.".format(obj))
+    else:
+        return obj
+
+# _callable_or_raise ()
+
+# -----------------------------------------------------------------------------
+class KeywordParserMixin(object):
+    """
+    Mixin providing additional Keywordparser specific parsing facilities.
+    """
+
+    @staticmethod
+    def _parse_arg_keys(arg_dict):
+        """
+        :param dict arg_dict: Dictionary like structure to be parsed
+
+        :returns: Tuple with argument keys
+        :rtype: dict
+        """
+
+        return tuple(arg_dict.keys())
+
+    # _parse_arg_keys ()
+
+    @staticmethod
+    def _parse_postfile(postfile):
+        """
+        Parse all argument keys from a POST request file.
+
+        :param str postfile: Postfile content
+
+        :returns: Tuple with parsed keys.
+        :rtype: tuple
+        """
+        argmap = {}
+
+        for line in postfile.split('\n'):
+            _line = line.split(
+                settings.FDSNWS_QUERY_VALUE_SEPARATOR_CHAR)
+            if len(_line) != 2:
+                continue
+
+            if all(w == '' for w in _line):
+                raise ValidationError('RTFM :)')
+
+            argmap[_line[0]] = _line[1]
+
+        return tuple(argmap.keys())
+
+    # _parse_postfile ()
+
+# class KeywordParserMixin
+
+
+class KeywordParser(KeywordParserMixin):
```

Comment:
@svmarti, I do not see the point why you should use the *mixin* approach, here. A `KeywordParser` should parse and validate keywords. This should be its default behaviour. That's why I suggest adding those `staticmethod`s to class `KeywordParser` in order to have a base `KeywordParser` with FDSNWS specific parsing facilities.